02-intermediate-ruby/debugging.md

```diff
@@ -0,0 +1,108 @@
+# Debugging
+
+## Learning Goals
+- Read and analyze a stack trace
+- Understand how to use `pry` for debugging in Ruby code
+- Better understand the scope of variables and instances in an application when testing
+
+### Reading Errors via Stack Traces
+
+When an exception is raised, it immediately stops the current method, just like `return`. It will then bubble its way up through the program, method by method. If it makes it all the way to the top, Ruby will print out a summary of the exception and a description of what the program was doing when the exception happened, also known as a **stack trace**.
```

Comment:
I feel like we are just dropped in here with no context - could you add a short introduction?